ui/components/src/LineChart/LineChart.tsx

```diff
@@ -191,7 +196,7 @@ export function LineChart({
       return __experimentalEChartsOptionsOverride(option);
     }
     return option;
-  }, [data, yAxis, unit, grid, legend, noDataOption, timeZone, __experimentalEChartsOptionsOverride]);
+  }, [data, yAxis, unit, grid, legend, noDataOption, timeZone, __experimentalEChartsOptionsOverride, noDataVariant]);
```

Comment:
This dependency array has almost as many items in it as the one TimeSeriesChartPanel now 🙃 
I hope we get more time to refactor both components soon!

Reply:
Saaaame. I have some thoughts for the `TimeSeriesPanel` and almost did a little breaking up in https://github.com/perses/perses/pull/1175, but realized it'd make the git diffs incomprehensible. I'm hoping I'll be able to come back and take a pass at it once I get the basic table legend PR out.